src/ai/backend/manager/container_registry/base.py

```diff
@@ -284,9 +288,15 @@ async def _scan_image(
                     )
         if (reporter := progress_reporter.get()) is not None:
             reporter.total_progress += len(tags)
-        async with aiotools.TaskGroup() as tg:
-            for tag in tags:
-                tg.create_task(self._scan_tag(sess, rqst_args, image, tag))
+
+        try:
+            async with aiotools.TaskGroup() as tg:
+                for tag in tags:
+                    tg.create_task(self._scan_tag(sess, rqst_args, image, tag))
+        except aiotools.TaskGroupError as e:
+            raise ScanImageError(
+                f"Image scan failed, Details: {cast(ExceptionGroup, e).exceptions}"
+            ) from e
```

Comment:
I'd like to remove this unnecessary casting.

Reply:
Since aiotools.TaskGroupError already has an exceptions field with type inheritance, Did you get an error if you don't cast it?

Reply:
Yes, it shows following error, though I don't know why it complains about it..

```
"TaskGroupError" has no attribute "exceptions"
```

Reply:
Looking into the aiotools code, I found that TaskGroupError conditionally inherits from ExceptionGroup to support legacy Python versions. For now, I'll just add a casting and move on.